### PR TITLE
Replace LGTM bot

### LINKED ISSUE #50: Handle LGTM and replace lgtm bot
The [lgtm bot](https://github.com/go-gitea/lgtm) is essentially unmaintained and has a number of bugs:

- It forgets approval status after the branch is updated via merge and needs to be triggered with a comment to update status, which is very annoying.
- It incorrectly parses "LGTM after ..." from comments

I think we could replace it with lightweight functionality in this bot. The bot tasks are:

- Handle `lgtm/need-*` labels, a subtraction of required approver count against actual approvers.
- Add a [commit status](https://docs.github.com/en/rest/commits/statuses?apiVersion=2022-11-28) or [check](https://docs.github.com/en/rest/checks?apiVersion=2022-11-28) that suceeds if approval count is >= 2. See [here](https://stackoverflow.com/questions/67919168/github-checks-api-vs-check-runs-vs-check-suites/72312617#72312617) for the diferences between the APIs.

Parsing comments is out of scope for this bot. It is simply not needed any more now that GH has the approvers data (the lgtm bot came from a time before this where a PR could only have a single approver).

For counting approvers, it should count maintainer role and above and this is where we differ from GitHub's view of approvers, where we allow even people without push access to approve and those are visible by the green checkmark. Maybe the [upcoming repo rules](https://github.blog/changelog/2023-04-17-introducing-repository-rules-public-beta/) will allow us to give maintainers a green checkmark.

### PR DESCRIPTION
**Pull request by yardenshoham**
This will do the same thing the LGTM bot did more or less. We won't parse the `MAINTAINERS` file nor look for comments containing the word LGTM.

This is experimental as I couldn't really test it. Before this is merged, the old LGTM bot must be stopped. After this is merged, the context `backporter/lgtm` must be added to `go-gitea/gitea`'s required status checks in all protected branches (and `approvals/lgtm` removed).
image

If the `BACKPORTER_GITHUB_TOKEN` used for production doesn't contain the `repo:status` scope, it must be added before merging.
image

- Closes #50

### CODE REVIEW

**Comment by silverwind** [comment]
After merge, we should disable lgtm bot so the two bots don't conflict.

File: src/github_test.ts
Comment (silverwind) [inline src/github_test.ts:29]: It's a bit dangerous to test against PRs that could still change, but I guess if we lock those PRs, it should be safe enough.

**Comment by yardenshoham** [comment]
Almost done with the logic, just gotta hook it up to the webhook listener

File: src/github.ts
Comment (yardenshoham) [reply src/github.ts:136]: That's what it's doing

Comment (lunny) [reply src/github.ts:136]: If one person rejects twice first and then approved once later, do you think it's approved or rejected?

Comment (yardenshoham) [reply src/github.ts:136]: approved

File: src/github_test.ts
Comment (yardenshoham) [reply src/github_test.ts:29]: Those PRs are closed, I don't think they will change. If you can, lock them

File: src/github.ts
Comment (lunny) [reply src/github.ts:136]: Yes, so is your algorithm still correct?

Comment (yardenshoham) [reply src/github.ts:136]: I think so. Where do you see an issue? On the first `CHANGES_REQUESTED` we delete the reviewer's name from the set (which wasn't there in the first place so we basically do nothing), on the second `CHANGES_REQUESTED` we, again delete the reviewer's name from the set (which isn't there anyway so we do nothing). On the `APPROVED` we add the reviewer's name to the set so it is counted finally. If you give me a PR number I can add it to the tests to make sure.

Comment (lunny) [reply src/github.ts:136]: If reviews is sorted by time asc, I think it's OK.

Another problem is we should just count official approvals. I remember anyone could approve in `go-gitea/gitea` repository.

Comment (yardenshoham) [reply src/github.ts:136]: It's sorted by time. From https://docs.github.com/en/rest/pulls/reviews?apiVersion=2022-11-28#list-reviews-for-a-pull-request:
> The list of reviews returns in chronological order.

All approvals in `go-gitea/gitea` are official. A normal user can't approve:
image

File: src/github_test.ts
Comment (silverwind) [reply src/github_test.ts:29]: locked them

File: src/github.ts
Comment (silverwind) [reply src/github.ts:136]: I think this must be some specific setting on the gitea repo. Normally anyone can approve any PR. I guess we can see how this goes. If we notice incorrect approvals, we can fix it later.

**Comment by yardenshoham** [comment]
Ready for review

**Review by wxiaoguang — APPROVED** [review]

File: src/lgtm.ts
Comment (silverwind) [inline src/lgtm.ts:68]: Maybe make this `2` a number constant and make the code below use it? Would be nice, in case we ever want to change this number to for example 3.

Comment (yardenshoham) [reply src/lgtm.ts:68]: That's quite a bit of logic (message templating) for something unlikely... If we want it to be 3 it's not a lot of effort (as opposed to the general case)

Comment (silverwind) [reply src/lgtm.ts:68]: For this function alone, it should be little work, but I guess you mean there's stuff outside the function that needs the number?

Comment (yardenshoham) [reply src/lgtm.ts:68]: I mean the message variable (uses two instead of 2)

Comment (silverwind) [reply src/lgtm.ts:68]: Ah, yes the "2" to "two" convertion.

**Review by silverwind — APPROVED** [review]

**Comment by yardenshoham** [comment]
@techknowlogick please read the PR description then merge

**Pull request merged**
